Shared/sdk/CVector4D.h

```diff
@@ -20,48 +20,38 @@
 class CVector4D
 {
 public:
-    CVector4D()
-    {
-        fX = 0;
-        fY = 0;
-        fZ = 0;
-        fW = 0;
-    }
+    float fX = 0.0f;
+    float fY = 0.0f;
+    float fZ = 0.0f;
+    float fW = 0.0f;
 
-    CVector4D(float _fX, float _fY, float _fZ, float _fW)
-    {
-        fX = _fX;
-        fY = _fY;
-        fZ = _fZ;
-        fW = _fW;
-    }
+    constexpr CVector4D() noexcept = default;
 
-    CVector4D(const CVector4D& vec)
-    {
-        fX = vec.fX;
-        fY = vec.fY;
-        fZ = vec.fZ;
-        fW = vec.fW;
-    }
+    constexpr CVector4D(const CVector4D&) noexcept = default;
+
+    constexpr CVector4D& operator=(const CVector4D&) noexcept = default;
 
-    CVector4D& operator=(const CVector4D& vec)
+    constexpr CVector4D(float x, float y, float z, float w) noexcept :
+        fX(x),
+        fY(y),
+        fZ(z),
+        fW(w)
     {
-        fX = vec.fX;
-        fY = vec.fY;
-        fZ = vec.fZ;
-        fW = vec.fW;
-        return *this;
     }
 
-    float DotProduct(CVector4D& other) const { return fX * other.fX + fY * other.fY + fZ * other.fZ; }
+    // Warning, this function is returning the wrong value(fW is missing), its kept because nothing uses it, only
+    // CLuaVector4DDefs.
+    constexpr float DotProduct(const CVector4D& other) const noexcept { return fX * other.fX + fY * other.fY + fZ * other.fZ; }
 
-    float Length() const { return sqrt(fX * fX + fY * fY + fZ * fZ + fW * fW); }
+    float Length() const noexcept { return sqrt(fX * fX + fY * fY + fZ * fZ + fW * fW); }
 
-    float LengthSquared() const { return (fX * fX) + (fY * fY) + (fZ * fZ) + (fW * fW); }
+    // LengthSquared returns Length() without sqrt applied (i.e. returns x*x* + y*y + z*z + w*w)
+    // For more info see CMaterialLine3DBatcher::Flush()
```

Comment:
```suggestion
    // LengthSquared returns Length() without sqrt applied (i.e. returns x*x* + y*y + z*z + w*w).
    // This can be useful if you only want to compare lengths.
```